packages/puppeteer-core/src/cdp/Connection.ts

```diff
@@ -117,6 +118,8 @@ export class Connection extends EventEmitter<CDPSessionEvents> {
     sessionId?: string,
     options?: CommandOptions
   ): Promise<ProtocolMapping.Commands[T]['returnType']> {
+    assert(!this.#closed, 'Protocol error: Connection closed.');
```

Comment:
so this turns out to actually be a breaking change, the errors can not be thrown synchronously

Reply:
I am gonna publish a fix